modules/renter/download.go

```diff
@@ -517,6 +517,27 @@ func (r *Renter) DownloadHistory() []modules.DownloadInfo {
 		} else {
 			downloads[i].Error = ""
 		}
+		i++
 	}
 	return downloads
 }
+
+// ClearDownloadHistory clears the renter's download history
+func (r *Renter) ClearDownloadHistory() {
+	r.downloadHistoryMu.Lock()
+	defer r.downloadHistoryMu.Unlock()
+	r.downloadHistory = make(map[string]*download)
+}
+
+// RemoveFromDownloadHistory removes a provided download from
+// the download history
+func (r *Renter) RemoveFromDownloadHistory(staticSiaPath string) error {
+	r.downloadHistoryMu.Lock()
+	defer r.downloadHistoryMu.Unlock()
+
+	delete(r.downloadHistory, staticSiaPath)
+	if _, ok := r.downloadHistory[staticSiaPath]; ok {
```

Comment:
I don't think that is necessary. I don't want to live in a world in which I can't trust `delete`.